tensorrt_llm/_torch/pyexecutor/model_engine.py

```diff
@@ -1684,7 +1726,9 @@ def previous_seq_slots_device():
         # so that we can update the kv_lens_cuda correctly in _preprocess_inputs.
         attn_metadata.num_chunked_ctx_requests = 0
         if self.enable_spec_decode and spec_config.spec_dec_mode.extend_ctx(
-                self.attn_backend):
+                self.attn_backend) and spec_config.is_linear_tree:
```

Comment:
Does it work on Hopper? The XQA kernels should work in theory but I haven't made all the tests pass for the linear tree case yet.